Sources/Subs.php

```diff
@@ -1041,11 +1038,11 @@ function get_date_or_time_format($type = '', $format = '')
 			// Anything that isn't a specification, punctuation mark, or whitespace.
 			'~(?<!%)\p{L}|[^\p{L}\p{P}\s]~u',
 			// A series of punctuation marks (except %), possibly separated by whitespace.
-			'~([^%\P{P}])(\s*)(?'.'>(\1|[^%\P{Po}])\s*(?!$))*~u',
+			'~([^%\P{P}])(\s*)(?' . '>(\1|[^%\P{Po}])\s*(?!$))*~u',
```

Comment:
Are we sure we want these changes?  Seems to change the nature of the regex...

Reply:
Ah...  Maybe OK?  Looks like an 'atomic group'.  Hadn't seen one of those before...

Reply:
This is fine. It's just adding some whitespace between concatenated parts of the regex. We concatenate the parts to make sure that `?>` in the regex will never be interpreted by PHP as a closing tag. (Normally it won't be because it is in a string, but if the line containing it is ever commented out, PHP suddenly dies a screaming death.)